File: app/src/main/java/com/starishko/pdd/RulesAdapter.java
package com.starishko.pdd;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import org.jetbrains.annotations.NotNull;

public class RulesAdapter extends RecyclerView.Adapter<RulesAdapter.RulesViewHolder> {

    @SuppressLint("StaticFieldLeak")
    private static Context parent;
    private int numberRules;
    RulesAdapter(int numberOfItems, Context parent){

        numberRules = numberOfItems;
        RulesAdapter.parent = parent;
    }

    @NonNull
    @Override
    public RulesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater= LayoutInflater.from(context);
        @SuppressLint("ResourceType")
        View view = inflater.inflate(R.layout.number_list_rule, parent, false);


        return new RulesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RulesViewHolder holder, final int position) {
        holder.bind(position);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(view.getContext(), Rule.class);
                intent.putExtra("numberTheme",  position+1);
                view.getContext().startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return numberRules;
    }

    static class RulesViewHolder extends RecyclerView.ViewHolder{
        TextView NumberThemeRule;

        RulesViewHolder(@NonNull View itemView) {
            super(itemView);
            NumberThemeRule = itemView.findViewById(R.id.number_list_rule);

        }

        void bind(int numberTheme){
            DatabaseReference database = FirebaseDatabase.getInstance().getReference();
            DatabaseReference myRefNumberTheme = database.child ("Rules").child (String.valueOf(numberTheme+1));

            myRefNumberTheme .addListenerForSingleValueEvent(new ValueEventListener() {
                @Override
                public void onDataChange(@NotNull DataSnapshot dataSnapshot) {
                    String value = dataSnapshot.child("Theme").getValue(String.class);
                    NumberThemeRule.setText(value);
                }
                @Override
                public void onCancelled(@NotNull DatabaseError databaseError) {

                }
            });


        }

    }
}

File: app/src/main/java/com/starishko/pdd/RulesAdapter.kt
package com.starishko.pdd

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.starishko.pdd.R
import android.widget.TextView
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import android.content.Intent
import com.starishko.pdd.RulesList
import android.annotation.SuppressLint
import android.widget.ProgressBar
import com.starishko.pdd.TicketsList
import android.content.SharedPreferences
import androidx.fragment.app.FragmentActivity
import com.starishko.pdd.Question.Postman
import com.starishko.pdd.Question
import android.widget.HorizontalScrollView
import android.app.Activity
import android.content.Context
import androidx.annotation.RequiresApi
import android.os.Build
import android.view.LayoutInflater
import android.view.ViewGroup
import com.google.firebase.storage.StorageReference
import com.google.firebase.storage.FirebaseStorage
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat
import com.google.android.gms.tasks.OnSuccessListener
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import com.google.android.gms.tasks.OnFailureListener
import android.widget.Toast
import com.starishko.pdd.DialogClearSettings
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.LinearLayoutManager
import com.starishko.pdd.RulesAdapter
import androidx.recyclerview.widget.GridLayoutManager
import com.starishko.pdd.TicketsAdapter
import com.starishko.pdd.RulesAdapter.RulesViewHolder
import com.starishko.pdd.TicketsAdapter.TicketsViewHolder
import androidx.appcompat.app.AppCompatDialogFragment
import android.content.DialogInterface
import android.view.View

class RulesAdapter internal constructor(private val numberRules: Int, parent: Context?) :
    RecyclerView.Adapter<RulesViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RulesViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        @SuppressLint("ResourceType") val view =
            inflater.inflate(R.layout.number_list_rule, parent, false)
        return RulesViewHolder(view)
    }

    override fun onBindViewHolder(holder: RulesViewHolder, position: Int) {
        holder.bind(position)
        holder.itemView.setOnClickListener { view ->
            val intent = Intent(view.context, Rule::class.java)
            intent.putExtra("numberTheme", position + 1)
            view.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return numberRules
    }

    class RulesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var NumberThemeRule: TextView
        fun bind(numberTheme: Int) {
            val database = FirebaseDatabase.getInstance().reference
            val myRefNumberTheme = database.child("Rules").child((numberTheme + 1).toString())
            myRefNumberTheme.addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onDataChange(dataSnapshot: DataSnapshot) {
                    val value = dataSnapshot.child("Theme").getValue(
                        String::class.java
                    )
                    NumberThemeRule.text = value
                }

                override fun onCancelled(databaseError: DatabaseError) {}
            })
        }

        init {
            NumberThemeRule = itemView.findViewById(R.id.number_list_rule)
        }
    }

    companion object {
        @SuppressLint("StaticFieldLeak")
        private var parent: Context? = null
    }

    init {
        Companion.parent = parent
    }
}